### ISSUE TITLE
Xamarin Android bug in Accessibility/TalkBack, where the Parent View Blocks Children

### ISSUE THREAD

**Issue by geeetarguy** [xamarin/Xamarin.Forms#6758]
### Description
There is a Xamarin Android bug in Accessibility/TalkBack, where the Parent View of a Container View will block all of its children from being accessed while in TalkBack. Look at the steps to reproduce, but what is being experienced is that a ListView's name is being called out and none of the child rows are able to be touched, selected or heard. 
 
iOS seems to be working just fine. This is limited to Android.

### Steps to Reproduce

1. ListView in xaml
2. ListView's .ItemSource and .ItemTemplate assigned
<StackLayout>
    <BoxView />
    <ListView x:Name="ListViewName" StyleId="ListViewName" CachingStrategy="RecycleElement" BackgroundColor="White" HorizontalOptions="FillAndExpand" VerticalOptions="FillAndExpand"  />
</StackLayout>

### Expected Behavior
Should be able to use TalkBack and access and hear the rows of the ListView without the parent's name being spoken.

### Actual Behavior
ListView's name is being spoken and all child rows are unreachable. Only the parent ListView is accessible.

### Basic Information

- Version with issue: 3.6.0.344457
- Last known good version: ???
- IDE: Visual Studio for Mac 2019
- Platform Target Frameworks: 
  - Android: Min 4.4, Target 9.0
- Android Support Library Version: 
- Nuget Packages: Xamarin.Forms
- Affected Devices: Android

### Screenshots

Unavailable. 

### Reproduction Link

See above xaml and instructions.

**Comment by samhouts** [comment]
@geeetarguy  Can you please attach a small project that demonstrates this issue? Thanks!

**Comment by geeetarguy** [comment]
Thanks for getting back to me, @samhouts Here is the code in question:


AndroidListViewTalkBackBug.xaml

<?xml version="1.0" encoding="UTF-8"?>
<ContentPage x:Name="AndroidListViewTalkBackBug" xmlns="http://xamarin.com/schemas/2014/forms" xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" xmlns:generic="clr-namespace:GenericNativeApp;assembly=GenericNativeApp" x:Class="GenericNativeApp.AndroidListViewTalkBackBug">
    <ContentPage.Content BackgroundColor="#CD2418">
        <Grid BackgroundColor="{StaticResource backgroundWhiteColor}" ColumnSpacing="0" RowSpacing="0" VerticalOptions="FillAndExpand" HorizontalOptions="Fill">
            <Grid.Padding>
                <OnPlatform x:TypeArguments="Thickness">
                    <OnPlatform.iOS>
                        0, 15, 0, 0
                    </OnPlatform.iOS>
                    <OnPlatform.Android>
                        0, 0, 0, 0
                    </OnPlatform.Android>
                </OnPlatform>
            </Grid.Padding>
            <Grid.RowDefinitions>
                <RowDefinition Height="Auto" />
                <RowDefinition Height="1" />
                <RowDefinition Height="*" />
            </Grid.RowDefinitions>
            <Grid.ColumnDefinitions>
                <ColumnDefinition Width="0.75*" />
                <ColumnDefinition Width="*" />
                <ColumnDefinition Width="0.75*" />
            </Grid.ColumnDefinitions>
            <ContentView x:Name="CancelButton" StyleId="CancelButton" Grid.Column="0" Grid.Row="0" HorizontalOptions="FillAndExpand" Padding="{1,1,0.5,1}" VerticalOptions="FillAndExpand">
                <Label FontSize="Small" HorizontalOptions="FillAndExpand" HorizontalTextAlignment="Start" Style="{StaticResource textLink}" Text="CANCEL" VerticalOptions="FillAndExpand" VerticalTextAlignment="Center" />
            </ContentView>
            <ContentView x:Name="PageTitle" Grid.Column="1" Grid.Row="0" BackgroundColor="{StaticResource backgroundWhiteColor}" HorizontalOptions="FillAndExpand" Padding="{Padding=0.5,1,0.5,1}" VerticalOptions="FillAndExpand">
                <Label HorizontalOptions="FillAndExpand" HorizontalTextAlignment="Center" Text="text" LineBreakMode="TailTruncation" VerticalOptions="FillAndExpand" VerticalTextAlignment="Center" />
            </ContentView>
            <ContentView x:Name="DoneButtonContent" Grid.Column="2" Grid.Row="0" BackgroundColor="{StaticResource backgroundWhiteColor}" HorizontalOptions="FillAndExpand" Padding="{Padding=0.5,1,1,1}" VerticalOptions="FillAndExpand">
                <Button x:Name="DoneButton" FontSize="Small" Command="{Binding SaveBrandFilter}" IsEnabled="{Binding SelectionsChanged}" Style="{StaticResource primaryButton}" StyleId="DoneButton" Text="DONE" />
            </ContentView>
            <BoxView Grid.Row="1" Grid.Column="0" Grid.ColumnSpan="3" Style="{StaticResource horizontalDivider}" />
            
            <StackLayout BackgroundColor="{StaticResource bodyBackgroundColor}" HorizontalOptions="FillAndExpand" VerticalOptions="FillAndExpand" Grid.Row="2" Grid.Column="0" Grid.ColumnSpan="3" Padding="{Padding=0,1,0,0}">
                <StackLayout IsVisible="{Binding ShowBody}" BackgroundColor="{StaticResource backgroundWhiteColor}" HorizontalOptions="FillAndExpand" VerticalOptions="FillAndExpand">
                    <BoxView Style="{StaticResource horizontalDivider}" />
                    
                    <ListView x:Name="RefinementsListContent" StyleId="RefinementsListContent" CachingStrategy="RecycleElement" BackgroundColor="White" HorizontalOptions="FillAndExpand" VerticalOptions="FillAndExpand" />
                </StackLayout>
            </StackLayout>
            
            <g:LoadingBlocker x:Name="LoadingIndicator" StyleId="LoadingIndicator" Grid.Row="2" Grid.Column="0" Grid.ColumnSpan="3" ShowLoading="{Binding ShowLoading}" />
        </Grid>
    </ContentPage.Content>
</ContentPage>



AndroidListViewTalkBackBug.xaml.cs:

 using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Threading.Tasks;
using GenericNativeApp.ViewModels;
using Xamarin.Forms;

namespace GenericNativeApp
{
    public partial class AndroidListViewTalkBackBug : ContentPage
    {
        ProductFinderViewModel ProductFinderVM;

        public BrandsSelectPage(ProductFinderViewModel vm, string pageType)
        {
            // set view model value to the passed in vm
            ProductFinderVM = vm;

            // set the binding context
            BindingContext = ProductFinderVM;

            //  set the page type
            ProductFinderVM.ProductFinderPageType = pageType;

            // init the view
            Init();

            // build brands listview
            buildBrands(ProductFinderVM);
        }

        void Init()
        {
            // page and nav init
            InitializeComponent();
            NavigationInit();

            // add event handlers
            AddEventHandlers();
        }

        void AddEventHandlers()
        {
            // cancel button
            var cancelTap = new TapGestureRecognizer();
            cancelTap.Tapped += async (s, e) =>
            {
                //	 pop page off nav stack
                await NavigationHelper.PopAsync();
            };
            CancelButton.GestureRecognizers.Add(cancelTap);
        }

        async void buildBrands(ProductFinderViewModel vm)
        {
            // load brands
            await vm.LoadOptions("brand");

            // set refinement list itemsource and template

            // This is what TalkBack 
            //says: "RefinementsListContent" and outlines the ListView without allowing touch to its child views, 
            //i.e rows
            RefinementsListContent.ItemsSource = vm.FacetingMethods; 
            RefinementsListContent.ItemTemplate = new DataTemplate(typeof(BrandsSelectRow));

            // pre-select facet if one had been previously selected
            if (!string.IsNullOrWhiteSpace(vm.BrandCommittedValue))
            {
                // search for selected item
                int index = vm.FacetingMethods.FindIndex(o => o.FacetValue == vm.BrandCommittedValue);

                // set selected item
                vm.FacetingMethods[index].IsSelected = true;

                // disable done button
                vm.SelectionsChanged = false;
            }

            // list view tap handling
            RefinementsListContent.ItemSelected += (sender, e) =>
            {
                // don't do anything if item was de-selected
                if (e.SelectedItem == null) { return; }

                // find last selected item
                int lastIndex = vm.FacetingMethods.FindIndex(o => o.IsSelected == true);

                // deselect last selected item
                if (lastIndex >= 0) { vm.FacetingMethods[lastIndex].IsSelected = false; }

                // search for selected item
                int index = vm.FacetingMethods.FindIndex(o => o.FacetValue == (e.SelectedItem as FacetingMethod).FacetValue);

                // set selected item
                vm.FacetingMethods[index].IsSelected = true;

                // loop through facet methods and display brand
                foreach (var facet in vm.FacetingMethods)
                {
                    if (facet.FacetValue == vm.FacetingMethods[index].FacetValue)
                    {
                        vm.BrandSelectedLabel = facet.DisplayValue;
                    }
                }

                if (vm.BrandCommittedValue == vm.FacetingMethods[index].FacetValue)
                {
                    // disable done button
                    vm.SelectionsChanged = false;
                }
                else
                {
                    // enable done button
                    vm.SelectionsChanged = true;
                }

                // set selected brand value
                vm.BrandSelectedValue = vm.FacetingMethods[index].FacetValue;

                // de-select other items
                ((ListView)sender).SelectedItem = null;
            };
        }
    }
}

**Comment by jsuarezruiz** [comment]
I attach the sample.
[Test6894.zip](https://github.com/xamarin/Xamarin.Forms/files/3411112/Test6894.zip)

**Comment by rmarinho** [comment]
Hello, i have tried to reproduce this issue with no luck, i tried this sample code

https://github.com/rmarinho/XFAssemblyInfo/blob/master/App9/App9/ViewModels/ItemsViewModel.cs

and talkback works ok, can you please change that sample to reproduce the problem and get back to us.

**Comment by samhouts** [comment]
@geeetarguy  Since we haven't heard from you in more than 30 days, we hope this issue is no longer affecting you. If it is, please reopen this issue and provide the requested information so that we can look into it further. Thank you!
